addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -483,9 +484,9 @@ def _l10n_in_edi_generate_invoice_json(self, invoice):
                     + tax_details_by_code.get("state_cess_non_advol_amount", 0.00)),
                 ),
                 "RndOffAmt": self._l10n_in_round_value(
-                    sum(line.balance for line in invoice.line_ids if line.display_type == 'rounding')),
+                    rounding_amount),
                 "TotInvVal": self._l10n_in_round_value(
-                    (tax_details.get("base_amount") + tax_details.get("tax_amount"))),
+                    (tax_details.get("base_amount") + tax_details.get("tax_amount") - rounding_amount)),
```

Comment:
@jva-odoo - can be for the balance being negative when the lines are positive, but what happens in the case of a refund?

Reply:
(it will probably just have the opposite balance as well...)

Reply:
let me check

Reply:
@jco-odoo We check and find there is a problem if round-off amount is negative
so we fix it. Can you re-check this

Thank you